test/Kestrel.Transport.Libuv.FunctionalTests/ListenHandleTests.cs

```diff
@@ -14,26 +14,30 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
     [OSSkipCondition(OperatingSystems.Windows, SkipReason = "Listening to open TCP socket and/or pipe handles is not supported on Windows.")]
     public class ListenHandleTests : LoggedTest
     {
+        // The Socket.Handle will be passed into libuv, which calls close() on the file descriptor when TestServer is disposed.  If
+        // the managed Socket is also disposed or finalized, it will try to call close() again on the file descriptor, which may lead to
+        // race condition bugs (test hangs) if the file descriptor has been re-used for another resource.  In .NET Core, objects
+        // assigned to static fields should never be disposed or finalized (even at process shutdown).
+        // https://github.com/aspnet/KestrelHttpServer/issues/2597
+        private static readonly Socket _listenSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
So this instance is usable for only one test?

Reply:
Yes, assigning the Socket to a static is a hacky way to prevent double-dispose.  I believe the less-hacky way would be to write our own cross-plat p-invokes to directly create a native Socket, but it's a lot more work for just test code.

Reply:
Name the field after the test then? Or embed it in the test?

Reply:
I renamed the field, not sure what you mean by "embed it in the test" since it needs to be a static.